packages/api/src/controllers/clip.ts

```diff
@@ -68,18 +56,36 @@ app.post("/", validatePost("clip-payload"), async (req, res) => {
         errors: ["Recording must be enabled on a live stream to create clips"],
       });
     }
-    session = await db.stream.getLastSessionFromSessionsTable(content.id);
-    const os = await db.objectStore.get(req.config.recordCatalystObjectStoreId);
-    url = pathJoin(os.publicUrl, session.playbackId, session.id, "output.m3u8");
+    let runningRecording = await getRunningRecording(content, req);
+    url = runningRecording.url;
+    session = runningRecording.session;
```

Comment:
Neat! :)

btw a ts tip is using deconstruction syntax (not requesting a change, just sharing if you like it)
```suggestion
    ({ url, session }) = await getRunningRecording(content, req);
```